### PR TITLE
Refactoring ProductCards

### PR DESCRIPTION
**Pull request by TuT597**
Prices are correctly displayed on every part of the website.

### CODE REVIEW

**Review by bartspreij — DISMISSED** [review]

**Comment by bartspreij** [comment]
Resolve merge conflict locally

**Review by bartspreij — CHANGES_REQUESTED** [review]
Resolve merge conflict locally

**Review by melleg — CHANGES_REQUESTED** [review]
ManageProducts is broken

**Review by bartspreij — APPROVED** [review]
Als je eslint regels uitzet dan liever bovenaan de file denk ik

File: frontend/src/components/discount/DiscountGallery.tsx [hunk 1]
```diff
@@ -1,33 +1,37 @@
...
+import Gallery from '../common/Gallery';
+import { Results, queryProducts } from '../../api/products-api';
 
 const DiscountGallery = () => {
-    const [discounts, setDiscounts] = useState<Product[]>([]);
+    const [results, setResults] = useState<Results>(new Results());
 
     useEffect(() => {
-        const loadDiscounts = async () => {
-            const fetchedDiscounts = await fetchDiscountProducts();
```
Comment (melleg) [inline frontend/src/components/discount/DiscountGallery.tsx:12]: This method should not be deleted

**Review by melleg — CHANGES_REQUESTED** [review]

File: frontend/src/components/discount/DiscountGallery.tsx (hunk 1 above)
Comment (TuT597) [reply frontend/src/components/discount/DiscountGallery.tsx]: Done

**Review by melleg — APPROVED** [review]

**Pull request merged**